doc/esmvalcore/config.rst

```diff
@@ -4,29 +4,163 @@
 Configuration files
 *******************
 
+Overview
+========
+
 There are several configuration files in ESMValTool:
 
-  - config-user.yml
-  - config-developer.yml
-  - config-references.yml
-  - config-logging.yml
+* ``config-user.yml``: sets a number of user-specific options like desired
+  graphical output format, root paths to data, etc.;
+* ``config-developer.yml``: sets a number of standardized file-naming and paths
+  to data formatting;
+* ``config-references.yml``: stores information on diagnostic authors and
+  scientific journals references;
+* ``config-logging.yml``: stores information on logging.
 
 User configuration file
 =======================
 
-See Section
+The ``config-user.yml`` is one of the two files the user needs to provide as
+input arguments to the ``esmvaltool`` executable at run time, the second being
+the :ref:`recipe`.
+
+The ``config-user.yml`` configuration file contains all the global level
+information needed by ESMValTool. It can be reused as many times the user needs
+to before changing any of the options stored in it. This file is essentially
+the gateway between the user and the machine-specific instructions to
+``esmvaltool``. The following shows the default settings from the
+``config-user.yml`` file with explanations in a commented line above each
+option:
+
+.. code-block:: yaml
+
+  # Diagnostics create plots? [true]/false
+  # turning it off will turn off graphical output from diagnostic
+  write_plots: true
+
+  # Diagnositcs write NetCDF files? [true]/false
+  # turning it off will turn off netCDF output from diagnostic
+  write_netcdf: true
+
+  # Set the console log level debug, [info], warning, error
+  # for much more information printed to screen set log_level: debug
+  log_level: info
+  # verbosity is deprecated and will be removed in the future
+  # verbosity: 1
+
+  # Exit on warning? true/[false]
+  exit_on_warning: false
+
+  # Plot file format? [ps]/pdf/png/eps/epsi
+  output_file_type: pdf
+
+  # Destination directory where all output will be written
+  # including log files and performance stats
+  output_dir: ./esmvaltool_output
+
+  # Auxiliary data directory (used for some additional datasets)
+  # this is where e.g. files can be downloaded to by a download
+  # script embedded in the diagnostic
+  auxiliary_data_dir: ./auxiliary_data
+
+  # Use netCDF compression true/[false]
+  compress_netcdf: false
+
+  # Save intermediary cubes in the preprocessor true/[false]
+  # set to true will save the output cube from each preprocessing step
+  # these files are numbered according to the preprocessing order
+  save_intermediary_cubes: false
+
+  # Remove the preproc dir if all fine
+  # this option true will remove ALL preprocessor files
```

Comment:
You want to make a full sentence out of this? Like "If this option is set to "true", it will remove ALL preprocessor files" or something.

Reply:
Done.